cis194/week2/tien/homework2/src/LogAnalysis.hs

```diff
@@ -0,0 +1,47 @@
+{-# OPTIONS_GHC -Wall #-}
+
+module LogAnalysis where
+
+import           Log
+import           Text.Read (readMaybe)
+
+getHead :: [String] -> String
+getHead wordList =
+  case wordList of
+    x:_ -> x
+    _   -> ""
+
+getNthValue :: Int -> String -> Maybe Int
+getNthValue n str = readMaybe (getHead . drop (n-1) . words $ str) :: Maybe Int
+
+getNthTailConent :: Int -> String -> String
+getNthTailConent n str = unwords . drop n . words $ str
+
+parseMaybeInt :: Maybe Int -> Int
+parseMaybeInt maybeValue =
+  case maybeValue of
+    Just i  -> i
+    Nothing -> 0
+
+parseMessage :: String -> LogMessage
+parseMessage logStr =
+  case getHead . words $ logStr of
+    "I" ->
```

Comment:
We can match on the first 2, 3 elements of `words logStr` here instead of just the first element. This way we can avoid using `getHead`, `getNthValue` and `getNthTailContent`.
e.g. :
```Haskell
case words logStr of
  ( "I" : ts : rest ) -> ....
  ( "W" : ts : rest ) -> ....
  ( "E" : lv : ts : rest ) -> ....
```